app/liquid/liquid_renderer.rb

```diff
@@ -11,7 +11,9 @@ def initialize(page, layout: nil, request_country: nil, member: nil, url_params:
   end
 
   def render
-    template.render( data ).html_safe
+    Rails.cache.fetch(cache_key) do
+      template.render( data ).html_safe
+    end
```

Comment:
this is cool! to clarify the relationship between this branch and my branch, if we benchmark it your branch will show all the speed improvements, but it will just totally screw up the use personalization until it has my changes. Is that right? do you want me to open my PR against dev or this branch?